src/centroiders.cpp

```diff
@@ -120,6 +118,127 @@ std::vector<Star> CenterOfGravityAlgorithm::Go(unsigned char *image, int imageWi
     return result;
 }
 
+//Determines how accurate and how much iteration is done by the IWCoG algorithm,
+//smaller means more accurate and more iterations.
+float iWCoGMinChange = 0.00001;
+
+struct IWCoGParams {
+    int xMin;
+    int xMax;
+    int yMin;
+    int yMax;
+    int cutoff;
+    int maxIntensity;
+    int guess;
+    std::unordered_set<int> checkedIndices;
+};
+
+void IWCoGHelper(IWCoGParams &p, int i, unsigned char *image, int imageWidth, int imageHeight, std::vector<int> &starIndices) {
+    if (i >= 0 && i < imageWidth * imageHeight && image[i] >= p.cutoff && p.checkedIndices.count(i) == 0) {
+        p.checkedIndices.insert(i);
+        starIndices.push_back(i);
+        if (image[i] > p.maxIntensity) {
+            p.maxIntensity = image[i];
+            p.guess = i;
+        }
+        if (i % imageWidth > p.xMax) {
+            p.xMax = i % imageWidth;
+        } else if (i % imageWidth < p.xMin) {
+            p.xMin = i % imageWidth;
+        }
+        if (i / imageWidth > p.yMax) {
+            p.yMax = i / imageWidth;
+        } else if (i / imageWidth < p.yMin) {
+            p.yMin = i / imageWidth;
+        }
+        if(i % imageWidth != imageWidth - 1) {
+            IWCoGHelper(p, i + 1, image, imageWidth, imageHeight, starIndices);
+        }
+        if (i % imageWidth != 0) {
+            IWCoGHelper(p, i - 1, image, imageWidth, imageHeight, starIndices);
+        }
+        IWCoGHelper(p, i + imageWidth, image, imageWidth, imageHeight, starIndices);
+        IWCoGHelper(p, i - imageWidth, image, imageWidth, imageHeight, starIndices);
+    }
+}
+
+std::vector<Star> IterativeWeightedCenterOfGravityAlgorithm::Go(unsigned char *image, int imageWidth, int imageHeight) const {
+    IWCoGParams p;
+    std::vector<Star> result;
+    p.cutoff = DetermineCutoff(image, imageWidth, imageHeight);
+    for (int i = 0; i < imageHeight * imageWidth; i++) {
+        //check if pixel is part of a "star" and has not been iterated over
+        if (image[i] >= p.cutoff && p.checkedIndices.count(i) == 0) {
+            std::vector<int> starIndices; //indices of the current star
+            p.maxIntensity = 0;
+            int xDiameter = 0; 
+            int yDiameter = 0;
+            float yWeightedCoordMagSum = 0; 
+            float xWeightedCoordMagSum = 0; 
+            float weightedMagSum = 0; 
+            float fwhm; //fwhm variable
+            float standardDeviation;
+            float w; //weight value
+            
+            //computes indices to skip after done w current star
+            int step = i;
```

Comment:
What's the point of this? We already check `p.checkedIndices.count(i) == 0` on line 171?

Reply:
done, also got rid of a simillar method in CoG